### PR TITLE
Added iframe instead of button in timathon history page for yt video

### PR DESCRIPTION
**Pull request by KrishnaKanth1729**
i can adjust width and height

### CODE REVIEW

File: TWT/apps/challenges/models/challenge.py [hunk 1]
```diff
@@ -70,3 +70,6 @@ def __str__(self):
     @property
     def active(self):
         return self.posted and not self.ended
+
+    def get_video_embed_link(self) -> str:
+        return f"https://youtube.com/embed/{self.youtube_video_link.split('=')[1]}"
```
Comment (takos22) [inline TWT/apps/challenges/models/challenge.py:75]: What for `youtu.be/...` links?

**Review by takos22 — CHANGES_REQUESTED** [review]

File: TWT/apps/challenges/models/challenge.py (hunk 1 above)
Comment (Shubhaankar-Sharma) [reply TWT/apps/challenges/models/challenge.py:75]: @takos22 its for the admin page :p may be we can skip that part, but sure we can resolve that, but for youtu.be links the regex is also going to fail in the model so we would also have to change that @KrishnaKanth1729

Comment (takos22) [reply TWT/apps/challenges/models/challenge.py:75]: The regex accepts `youtu.be` links, that's why I asked about it

https://github.com/Tech-With-Tim/twtcodejam.net/blob/dev/TWT/apps/challenges/models/challenge.py#L65

Comment (Shubhaankar-Sharma) [reply TWT/apps/challenges/models/challenge.py:75]: oh, yea then thats an issue

Comment (KrishnaKanth1729) [reply TWT/apps/challenges/models/challenge.py:75]: ok so u want me to remove the youtu.be part?

**Review by takos22 — APPROVED** [review]
LGTM, waiting for another approving review to merge

**Review by takos22 — APPROVED** [review]

**Comment by KrishnaKanth1729** [comment]
is everything fine @takos22 or any further changes are required. bcos i recall Shubhaankar-Sharma saying something abt it not being responsive

File: TWT/templates/timathon/codejam_listView.html [hunk 2]
```diff
@@ -20,7 +20,7 @@ <h6 class="text-muted card-subtitle mb-2" style="text-align: center;">{{challeng
                 <p class="card-text" style="text-align: center;">{{ challenge.challenge.short_desc }}</p>
                 {% if challenge.challenge.youtube_video_link %}
                     <div class="text-center">
-                        <a type="button" class="btn btn-info" href="{{ challenge.challenge.youtube_video_link }}" target="_blank">Youtube Video</a>
+                        <iframe width="90%" height="554" src="{{ challenge.challenge.get_video_embed_link }}" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```
Comment (mohamed040406) [inline TWT/templates/timathon/codejam_listView.html:23]: i don't think `90%` is a good "responsive" solution

**Review by mohamed040406 — CHANGES_REQUESTED** [review]

File: TWT/templates/timathon/codejam_listView.html (hunk 2 above)
Comment (KrishnaKanth1729) [reply TWT/templates/timathon/codejam_listView.html:23]: :hmmGe:

Comment (KrishnaKanth1729) [reply TWT/templates/timathon/codejam_listView.html:23]: what do u suggest i do?
media queries ?

**Comment by KrishnaKanth1729** [comment]
Screenshot (104)
this is how it looks on a small screen

**Comment by KrishnaKanth1729** [comment]
Screenshot (105)
Screenshot (106)
some more images :mhm:

**Comment by KrishnaKanth1729** [comment]
i have also added an orange border and a .5rem border-radius

**Comment by KrishnaKanth1729** [comment]
do let me know if any changes are required

**Pull request merged**